### PR TITLE
Bug 1794606: Use polymorphic helpers from exec, when reading a pod

### PR DESCRIPTION
**Pull request by soltysh**
Additionally, this deprecates special case `--timeout` flag in `oc rsh` in favour of the global `--request-timeout` flag.

/assign @sallyom

### CODE REVIEW

File: pkg/cli/rsh/rsh.go [hunk 1]
```diff
@@ -3,38 +3,27 @@ package rsh
...
-	RshRecommendedName = "rsh"
-	DefaultShell       = "/bin/sh"
+	RshRecommendedName   = "rsh"
+	DefaultShell         = "/bin/sh"
+	defaultPodRshTimeout = 60 * time.Second
 )
 
 var (
+	rshUsageStr    = "rsh (POD | TYPE/NAME) COMMAND [ARG1] [ARG2] ... [ARGN]"
+	rshUsageErrStr = fmt.Sprintf("expected '%s'.\nPOD or TYPE/NAME is a required argument for the logs command", rshUsageStr)
```
Comment (sallyom) [inline pkg/cli/rsh/rsh.go]: `for the rsh command`   : )

File: pkg/cli/rsh/rsh.go [hunk 2]
```diff
@@ -103,27 +93,35 @@ func NewCmdRsh(name string, parent string, f kcmdutil.Factory, streams genericcl
...
-	cmd.Flags().BoolVarP(&options.DisableTTY, "no-tty", "T", false, "Disable pseudo-terminal allocation")
-	cmd.Flags().StringVar(&options.Executable, "shell", DefaultShell, "Path to the shell command")
-	cmd.Flags().IntVar(&options.Timeout, "timeout", 10, "Request timeout for obtaining a pod from the server; defaults to 10 seconds")
-	cmd.Flags().StringVarP(&options.ContainerName, "container", "c", "", "Container name; defaults to first container")
+	kcmdutil.AddPodRunningTimeoutFlag(cmd, defaultPodRshTimeout)
+	cmd.Flags().BoolVarP(&options.ForceTTY, "tty", "t", options.ForceTTY, "Force a pseudo-terminal to be allocated")
+	cmd.Flags().BoolVarP(&options.DisableTTY, "no-tty", "T", options.DisableTTY, "Disable pseudo-terminal allocation")
+	cmd.Flags().StringVar(&options.Executable, "shell", options.Executable, "Path to the shell command")
+	cmd.Flags().IntVar(&options.Timeout, "timeout", options.Timeout, "Request timeout for obtaining a pod from the server; defaults to 10 seconds")
+	cmd.Flags().MarkDeprecated("timeout", "use --request-timeout, instead.")
```
Comment (Miciah) [inline pkg/cli/rsh/rsh.go]: Shouldn't that be "use --pod-running-timeout"?

Comment (soltysh) [reply pkg/cli/rsh/rsh.go]: Nope, this timeout was using when getting the pod in the old machanism.

**Pull request merged**